apps/meteor/client/views/omnichannel/contactInfo/ContactInfo/ReviewContactModal.tsx

```diff
@@ -0,0 +1,156 @@
+import { useCustomFieldsMetadata } from '../../directory/hooks/useCustomFieldsMetadata';
+
+const fieldNameMap: { [key: string]: TranslationKey } = {
+	name: 'Name',
+	contactManager: 'Contact_Manager',
+};
+
+function mapConflicts(
+	contact: Serialized<ILivechatContact>,
+	metadata: CustomFieldMetadata[],
+): Record<string, { name: string; label: string; values: string[] }> {
+	const conflicts = contact.conflictingFields?.reduce(
+		(acc, current) => {
+			const fieldName = current.field === 'manager' ? 'contactManager' : current.field.replace('customFields.', '');
+
+			if (acc[fieldName]) {
+				acc[fieldName].values.push(current.value);
+			} else {
+				acc[fieldName] = {
+					name: fieldName,
+					label:
+						(current.field.startsWith('customFields.') && metadata.find(({ name }) => name === fieldName)?.label) ||
+						fieldNameMap[fieldName],
+					values: [current.value],
+				};
+			}
+			return acc;
+		},
+		{} as Record<string, { name: string; label: string; values: string[] }>,
+	);
+
+	if (conflicts?.name?.values.length && contact.name) {
+		conflicts.name.values.push(contact.name);
+	}
+
+	if (conflicts?.contactManager?.values.length && contact.contactManager) {
+		conflicts.contactManager.values.push(contact.contactManager);
+	}
+
+	return conflicts || {};
+}
+
+type ReviewContactModalProps = {
+	contact: Serialized<ILivechatContact>;
+	onCancel: () => void;
+};
+
+type HandleConflictsPayload = {
+	name: string;
+	contactManager: string;
+	[key: string]: string;
+};
+
+const ReviewContactModal = ({ contact, onCancel }: ReviewContactModalProps) => {
+	const t = useTranslation();
+	const dispatchToastMessage = useToastMessageDispatch();
+	const hasLicense = useHasLicenseModule('livechat-enterprise');
+
+	const updateContact = useEndpoint('POST', '/v1/omnichannel/contacts.update');
+	const queryClient = useQueryClient();
+
+	const {
+		control,
+		handleSubmit,
+		formState: { errors },
+	} = useForm<HandleConflictsPayload>();
+
+	const canViewCustomFields = useAtLeastOnePermission(['view-livechat-room-customfields', 'edit-livechat-room-customfields']);
+
+	const { data: customFieldsMetadata = [] } = useCustomFieldsMetadata({
+		scope: 'visitor',
+		enabled: canViewCustomFields,
+	});
+
+	const handleConflicts = async ({ name, contactManager, ...customField }: HandleConflictsPayload) => {
+		const payload = {
+			name,
+			contactManager,
+			customFields: { ...customField },
+			wipeConflicts: true,
+		};
+
+		try {
+			await updateContact({ contactId: contact?._id, ...payload });
+			dispatchToastMessage({ type: 'success', message: t('Contact_has_been_updated') });
+			await queryClient.invalidateQueries(['getContactById']);
+		} catch (error) {
+			dispatchToastMessage({ type: 'error', message: error });
+		} finally {
+			onCancel();
+		}
+	};
+
+	const mappedConflicts = mapConflicts(contact, customFieldsMetadata);
+	const conflictingFields = Object.values(mappedConflicts);
```

Comment:
again I think this could be inside the use query, or a select function.
this results are not stable we are not memoizing nothing here